### ISSUE TITLE
Exclude objectives with (the) high(est) priority from being tasked as missions by AI commanders

### ISSUE THREAD

**Issue by Asmodeuz** [ALiVEOS/ALiVE.OS#542]
ALiVE wiki suggests to use high objective priority for a fixed logistics insertion point like so: _"Tip: use Military Placement (Custom Objectives) set to very high priority (e.g. 999) to set a custom fixed insertion point."_ (see: http://alivemod.com/wiki/index.php/Military_Logistics)

I've personally taken that suggestion to good use since by doing just that one can create a sorts of realistic reinforcement flow (from edges of the map for example) which is especially good for WW2 scenarios.
For a as-lenghty-as-possible mission reinforcements should keep flowing uninterrupted from these fixed insertion points and this in turn means that the AI commanders should never task a mission that involves attacking or capturing such objectives.

The problem is that the autotasking functionality **always** creates  _MilAssault_ or _MilDefend_ missions to objectives with 999 (or higher than any other) priority objectives and this usually happens already at the beginning of a mission.

Could the mod introduce a way for the mission makers to exclude objectives with high priority from being tasked as missions by AI commander? Or might there already be some other way to do just that (mind you that blacklisting the aforementioned tasks is out of the question since I'm here trying to get them incorporated into the missions I'm creating)?

**Comment by HayesRus** [comment]
Just dont sync "Custom Objective" with opposite AI commander if I understood you correctly.

**Comment by Asmodeuz** [comment]
That's the thing, I don't sync these _Custom objective_ modules with opposite AI commander. In fact I never have since I've used the custom objective module for quite some time as fixed insertion points in my missions.

See the picture below where I've created a mission with fixed insertion points for both the BLUFOR and OPFOR sides. Both the OPFOR and BLUFOR custom objective modules are only synced to a friendly _Military AI Commander_ module. 
![alive_custom_log_spawns](https://user-images.githubusercontent.com/12435032/45589768-6a709080-b934-11e8-9cdc-ec84b0fd613b.jpg)

Anyone can test the outcome with this mission [C2ISTAR_errors_CBA_ALIVE.Altis.zip](https://github.com/ALiVEOS/ALiVE.OS/files/2386026/C2ISTAR_errors_CBA_ALIVE.Altis.zip).

In that mission the player is on the BLUFOR side and will be tasked either with an assault mission where the target is the **OPF_Log_spawn** or a defend mission where the target is the **BLUF_Log_spawn**. Either of the missions might not get tasked immediately after the first spawn and you might receive _Capture objective_ mission(s) at first. But accomplish these and you'll eventually be tasked with the aforementioned assault or defend mission which _should not happen_ since the Custom objectives are not connected to their "enemy faction commanders".

**Comment by the-star** [comment]
Think one approach to solving this issue is

- Allow user to set a variable on an objective so they will be ignored by tasking

and/or

- Do not assign a task if opposing OPCOMs are not synced to that objective

Note, when I refer to tasks above I am referring to the C2ISTAR module which assigns player tasks.

**Comment by Asmodeuz** [comment]
the-star: I was under the impression that the second approach you mention is already implemented. And it would seem that HayesRus thought so too.

Any actually working approach would suit me that's for sure!

**Comment by the-star** [comment]
@Asmodeuz AI tasks should work properly (per the wiki). Player tasks does not as it uses different information for some reason. Note my emphasis on AI vs player tasks.

The MilAssault player task only considers whether an enemy OPCOM is holding an objective. Other player tasks have a similar problem.

Some relevant code:
https://github.com/ALiVEOS/ALiVE.OS/blob/755459f6bd166c52e6f978526218a951c2930857/addons/mil_c2istar/tasks/fnc_taskMilAssault.sqf#L66

https://github.com/ALiVEOS/ALiVE.OS/blob/755459f6bd166c52e6f978526218a951c2930857/addons/mil_c2istar/tasks/fnc_taskMilDefence.sqf#L66

**Comment by the-star** [comment]
@Asmodeuz would you be willing to test out my fork? 

You can download it [here](https://github.com/the-star/ALiVE.OS/releases/tag/v0.1-exclude_milcustom_obj). It contains a potential fix for this issue and the other issue you opened.

Would appreciate feedback on the following:
1. Does it work without script errors
1. Does the new parameter on the custom military objective make sense
1. General feedback

**Comment by Asmodeuz** [comment]
For sure I am willing and eager to test but I don't have the slightest clue of how to get your fork compatible with Arma 3. I guess I would need to pack all the folders in Addons into separate PBO's and I'd probably know how to do that. But isn't it so that the PBO's need to be .bisign'ed (no idea how to do that) and to be able to test in a dedicated server one needs a .bikey (no idea how to get that).

So, in short: willing to test but going to need some help on the way

**Comment by Asmodeuz** [comment]
Thank you the-star!

I finally got to test your fork and your approach worked well with the few tests I ran through the in-game editor. I also tested the changes you made in a dedicated server but only a couple of times because of issue https://github.com/ALiVEOS/ALiVE.OS/issues/552
"Thanks" to that issue autogenerated tasks stop from being automatically generated so only way to make ALiVE to offer you new tasks (after completing one that gets tasked at the start of a mission) is to restart the mission again and again. That's easier to do from the in-game editor but not in a dedicated server.

To me at least the new parameter in the custom objective module is self-explanatory but that could be because I know what we are talking about here.

All in all I did not observe any script errors or the like so IMHO this change is good to go. Though of that I'm not sure of if this is deemed as a "needed feature for ALiVE" :)
